{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "R[write to console]: ✔ ggplot2 3.3.6     ✔ purrr   0.3.4\n",
      "✔ tibble  3.1.7     ✔ dplyr   1.0.9\n",
      "✔ tidyr   1.2.0     ✔ stringr 1.4.0\n",
      "✔ readr   2.1.2     ✔ forcats 0.5.1\n",
      "\n",
      "R[write to console]: ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "library(tidyverse)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(broom)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "# Load data\n",
    "\n",
    "The goal here is to determine the if certain environments are enriched for microcins. To do that we need information about the assemblies so that we can determine which environments they came from.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New names:\n",
      "• `` -> `...4`\n",
      "• `` -> `...13`\n",
      "• `` -> `...14`\n",
      "• `` -> `...22`\n",
      "• `` -> `...32`\n",
      "• `` -> `...37`\n",
      "• `` -> `...38`\n",
      "• `` -> `...39`\n",
      "• `` -> `...40`\n",
      "• `` -> `...41`\n",
      "• `` -> `...42`\n",
      "• `` -> `...50`\n",
      "• `` -> `...57`\n",
      "• `` -> `...65`\n",
      "• `` -> `...71`\n",
      "• `` -> `...74`\n",
      "• `` -> `...75`\n",
      "• `` -> `...76`\n",
      "• `` -> `...80`\n",
      "• `` -> `...83`\n",
      "• `` -> `...84`\n",
      "# A tibble: 1,294 × 2\n",
      "   Genome_ID       phylogroup\n",
      "   <chr>           <chr>     \n",
      " 1 ESCO.0216.00064 A         \n",
      " 2 ESCO.0216.00065 B2        \n",
      " 3 ESCO.0216.00066 B2        \n",
      " 4 ESCO.0216.00067 B2        \n",
      " 5 ESCO.0216.00068 B2        \n",
      " 6 ESCO.0216.00069 B2        \n",
      " 7 ESCO.0216.00070 B2        \n",
      " 8 ESCO.0216.00071 B1        \n",
      " 9 ESCO.0216.00072 B2        \n",
      "10 ESCO.0216.00073 D         \n",
      "# … with 1,284 more rows\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "Australian_Dataset <- read_excel(\"data/Ecoli_assemblySummary.xlsx\", sheet=\"Australian-Dataset\", skip=4) %>%\n",
    "    select(c(\"Genome_ID\",\"Phylogroup\"))%>%\n",
    "\trename(\"phylogroup\"=\"Phylogroup\")%>%\n",
    "    drop_na()\n",
    "\n",
    "Australian_Dataset "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New names:\n",
      "• `` -> `...8`\n",
      "• `` -> `...20`\n",
      "• `` -> `...28`\n",
      "• `` -> `...35`\n",
      "# A tibble: 1,302 × 4\n",
      "   Genome_ID       `Isolation source` `Strain Category`      phylogroup\n",
      "   <chr>           <chr>              <chr>                  <chr>     \n",
      " 1 ESCO.0216.00064 Host-associated    Mammal_Faecal          A         \n",
      " 2 ESCO.0216.00065 Environmental      Water                  B2        \n",
      " 3 ESCO.0216.00066 Host-associated    Human_Extra-intestinal B2        \n",
      " 4 ESCO.0216.00067 Host-associated    Human_Extra-intestinal B2        \n",
      " 5 ESCO.0216.00068 Host-associated    Mammal_Faecal          B2        \n",
      " 6 ESCO.0216.00069 Host-associated    Human_Extra-intestinal B2        \n",
      " 7 ESCO.0216.00070 Environmental      Water                  B2        \n",
      " 8 ESCO.0216.00071 Environmental      Water                  B1        \n",
      " 9 ESCO.0216.00072 Host-associated    Human_Intestinal       B2        \n",
      "10 ESCO.0216.00073 Host-associated    Human_Intestinal       D         \n",
      "# … with 1,292 more rows\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "assemblySummary_SequencedGenomes <- read_excel(\"data/Ecoli_assemblySummary.xlsx\", \n",
    "                                               sheet=\"Sequenced-Genomes\", skip=5) %>%\n",
    "\tselect(c(\"Genome_ID\",\"Isolation source\",\"Strain Category\")) %>%\n",
    "    drop_na()%>%\n",
    "#Remove 'Alone' and 'Soil' samples\n",
    "\tfilter(`Strain Category` != \"Alone\") %>%\n",
    "\tfilter(`Strain Category` != \"Soil\")%>%\n",
    "#Removing three further samples that were categorized by Touchon as Alone\n",
    "    filter(Genome_ID != \"ESCO.0216.00100\") %>%\n",
    "    filter(Genome_ID != \"ESCO.0216.00262\") %>%\n",
    "    filter(Genome_ID != \"ESCO.0216.00275\") %>%\n",
    "#adding in phylogenetic data from above\n",
    "    full_join(Australian_Dataset, by = \"Genome_ID\") #%>%\n",
    "#    drop_na()\n",
    "\n",
    "assemblySummary_SequencedGenomes #has missing two"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 7537 Columns: 25\n",
      "── Column specification ────────────────────────────────────────────────────────\n",
      "Delimiter: \",\"\n",
      "chr  (8): qseqid, sseqid, component, cinful_id, pephash, sample, contig, seq\n",
      "dbl (13): pident, length, mismatch, gapopen, qstart, qend, sstart, send, eva...\n",
      "lgl  (4): hmmerHit, verified, allStandardAA, signalMatch\n",
      "\n",
      "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
      "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "# A tibble: 286 × 25\n",
      "   qseqid        sseqid pident length mismatch gapopen qstart  qend sstart  send\n",
      "   <chr>         <chr>   <dbl>  <dbl>    <dbl>   <dbl>  <dbl> <dbl>  <dbl> <dbl>\n",
      " 1 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 2 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 3 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 4 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 5 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 6 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 7 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 8 v1_PLS_db490… H47_s…  100       75        0       0     23    97      1    75\n",
      " 9 v1_PLS_7e75d… I47_t…   93.0     57        4       0      1    57      1    57\n",
      "10 v1_PLS_22016… V_sp_…   26.9    104       50       3      1    86      1    96\n",
      "# … with 276 more rows, and 15 more variables: evalue <dbl>, bitscore <dbl>,\n",
      "#   component <chr>, hmmerHit <lgl>, verified <lgl>, ASSEMBLY_ACC <chr>,\n",
      "#   pephash <chr>, sample <chr>, contig <chr>, start <dbl>, stop <dbl>,\n",
      "#   strand <dbl>, allStandardAA <lgl>, seq <chr>, signalMatch <lgl>\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "best_hits <- read_csv(\"data/best_hits_Touchon2020_2022.08.15.csv\")\n",
    "\n",
    "best_hits_assemblyID <- best_hits %>%\n",
    "\tseparate(cinful_id,c(\"ASSEMBLY_ACC_ID\"), sep=\"/\") %>%\n",
    "\tseparate(ASSEMBLY_ACC_ID, c(\"ASSEMBLY_ACC\"), sep = \"[.]\") %>%\n",
    "    distinct(ASSEMBLY_ACC, .keep_all = TRUE)%>%\n",
    "\tfilter(component == \"microcins.verified\") %>%\n",
    "    filter (hmmerHit == TRUE) %>%\n",
    "    drop_na()\n",
    "\n",
    "best_hits_assemblyID #286 genomes with microcin hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 1,213 × 9\n",
      "   Genome_ID       `Isolation source` `Strain Category` phylogroup ASSEMBLY_NAME\n",
      "   <chr>           <chr>              <chr>             <chr>      <chr>        \n",
      " 1 ESCO.0216.00064 Host-associated    Mammal_Faecal     A          M919         \n",
      " 2 ESCO.0216.00065 Environmental      Water             B2         E5795        \n",
      " 3 ESCO.0216.00066 Host-associated    Human_Extra-inte… B2         H219         \n",
      " 4 ESCO.0216.00069 Host-associated    Human_Extra-inte… B2         H144         \n",
      " 5 ESCO.0216.00070 Environmental      Water             B2         E7727        \n",
      " 6 ESCO.0216.00071 Environmental      Water             B1         E2508        \n",
      " 7 ESCO.0216.00072 Host-associated    Human_Intestinal  B2         54_1_Ti6     \n",
      " 8 ESCO.0216.00073 Host-associated    Human_Intestinal  D          47_4_R1      \n",
      " 9 ESCO.0216.00074 Host-associated    Mammal_Faecal     B2         M714         \n",
      "10 ESCO.0216.00075 Environmental      Water             B2         E7615        \n",
      "# … with 1,203 more rows, and 4 more variables: ASSEMBLY_ACC <chr>,\n",
      "#   STUDY_ID <chr>, SAMPLE_ID <chr>, CONTIG_ACC <chr>\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "\n",
    "assemblySummary_ENAInformation <- read_excel(\"data/Ecoli_assemblySummary.xlsx\", sheet=\"ENA-Information\") %>%\n",
    "    rename(\"Genome_ID\" = \"GEMBASE_NAME\")\n",
    "\n",
    "assemblySummary_Annotated <- assemblySummary_SequencedGenomes %>%\n",
    "\tright_join(assemblySummary_ENAInformation, by=\"Genome_ID\") %>%\n",
    "\tdrop_na()\n",
    "\n",
    "assemblySummary_Annotated #%>% #1213 genomes\n",
    "#\twrite_csv(\"data/supplemental/assemblySummary_Annotated.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Joining, by = \"ASSEMBLY_ACC\"\n",
      "# A tibble: 284 × 5\n",
      "   ASSEMBLY_ACC  phylogroup sseqid                     verified `Strain Catego…`\n",
      "   <chr>         <chr>      <chr>                      <lgl>    <chr>           \n",
      " 1 GCA_902858915 F          H47_sp_P62530_MCHB_ECOLX   FALSE    Human_Intestinal\n",
      " 2 GCA_902846155 B1         H47_sp_P62530_MCHB_ECOLX   FALSE    Human_Intestinal\n",
      " 3 GCA_902849625 G          H47_sp_P62530_MCHB_ECOLX   FALSE    Human_Faecal    \n",
      " 4 GCA_902840795 D          H47_sp_P62530_MCHB_ECOLX   FALSE    Mammal_Faecal   \n",
      " 5 GCA_902840495 B1         H47_sp_P62530_MCHB_ECOLX   FALSE    Mammal_Faecal   \n",
      " 6 GCA_902844105 A          H47_sp_P62530_MCHB_ECOLX   FALSE    Human_Faecal    \n",
      " 7 GCA_902846895 B1         H47_sp_P62530_MCHB_ECOLX   FALSE    Water           \n",
      " 8 GCA_902842745 B1         H47_sp_P62530_MCHB_ECOLX   FALSE    Water           \n",
      " 9 GCA_902844605 A          I47_tr_Q712Q0_Q712Q0_ECOLX FALSE    Human_Extra-int…\n",
      "10 GCA_902849245 G          V_sp_P22522_CEAV_ECOLX     FALSE    Human_Extra-int…\n",
      "# … with 274 more rows\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "best_hits_assemblyID_Annotated <- best_hits_assemblyID %>% #this contains 286 microcin hits\n",
    "\tleft_join(assemblySummary_Annotated) %>% #this contains 1213 genomes with other information\n",
    "\tselect(ASSEMBLY_ACC, phylogroup, sseqid, verified, `Strain Category`) %>%\n",
    "    drop_na(c(`Strain Category`, verified))\n",
    "\n",
    "\n",
    "best_hits_assemblyID_Annotated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 284\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "hits_phylogroup_by_category <- table(best_hits_assemblyID_Annotated$phylogroup, best_hits_assemblyID_Annotated$`Strain Category`)\n",
    "\n",
    "table(best_hits_assemblyID_Annotated$phylogroup, best_hits_assemblyID_Annotated$`Strain Category`)\n",
    "\n",
    "sum(table(best_hits_assemblyID_Annotated$phylogroup, best_hits_assemblyID_Annotated$`Strain Category`))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    \n",
      "     Bird_Faecal Human_Extra-intestinal Human_Faecal Human_Intestinal\n",
      "  A           51                     19           35               20\n",
      "  B1          46                      1            0               28\n",
      "  B2          31                     64           22               73\n",
      "  C            1                      5            4                1\n",
      "  D           29                      0            0               44\n",
      "  E            8                      0            0                1\n",
      "  F           17                     13            7                5\n",
      "  G            2                      2            4                0\n",
      "    \n",
      "     Mammal_Faecal Poultry_Meat Water\n",
      "  A             21           89    56\n",
      "  B1            27           48   123\n",
      "  B2            35           37    38\n",
      "  C              0            4     1\n",
      "  D             27           43    31\n",
      "  E              4           28    15\n",
      "  F              3           17     9\n",
      "  G              0           17     7\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "all_phylogroup_by_category <- table(assemblySummary_Annotated$phylogroup, assemblySummary_Annotated$`Strain Category`)\n",
    "\n",
    "table(assemblySummary_Annotated$phylogroup, assemblySummary_Annotated$`Strain Category`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 284\n",
      "[1] \"/\"\n",
      "[1] 1213\n",
      "[1] \"=\"\n",
      "[1] 0.2341303\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Frequency of microcin per assembly\n",
    "microcinAssemblyProportion <- length(best_hits_assemblyID_Annotated$ASSEMBLY_ACC)/length(assemblySummary_Annotated$ASSEMBLY_ACC)\n",
    "\n",
    "print(length(best_hits_assemblyID_Annotated$ASSEMBLY_ACC))\n",
    "print(\"/\")\n",
    "print(length(assemblySummary_Annotated$ASSEMBLY_ACC))\n",
    "print(\"=\")\n",
    "print(microcinAssemblyProportion)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 7 × 2\n",
      "  `Strain Category`      total_assemblies\n",
      "  <chr>                             <int>\n",
      "1 Bird_Faecal                         185\n",
      "2 Human_Extra-intestinal              104\n",
      "3 Human_Faecal                         72\n",
      "4 Human_Intestinal                    172\n",
      "5 Mammal_Faecal                       117\n",
      "6 Poultry_Meat                        283\n",
      "7 Water                               280\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "strain_category_assemblies <- assemblySummary_Annotated %>%\n",
    "\tcount(`Strain Category`, name=\"total_assemblies\")\n",
    "strain_category_assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Joining, by = \"Strain Category\"\n",
      "# A tibble: 7 × 6\n",
      "  `Strain Category`    microcin total_assemblies relative_microc…  odds      LOE\n",
      "  <chr>                   <int>            <int>            <dbl> <dbl>    <dbl>\n",
      "1 Bird_Faecal                42              185           0.227  0.970 -0.0134 \n",
      "2 Human_Extra-intesti…       46              104           0.442  1.89   0.276  \n",
      "3 Human_Faecal               17               72           0.236  1.01   0.00366\n",
      "4 Human_Intestinal           33              172           0.192  0.819 -0.0865 \n",
      "5 Mammal_Faecal              26              117           0.222  0.949 -0.0227 \n",
      "6 Poultry_Meat               94              283           0.332  1.42   0.152  \n",
      "7 Water                      26              280           0.0929 0.397 -0.402  \n"
     ]
    }
   ],
   "source": [
    "%%R -w 1500\n",
    "\n",
    "microcin_strain_category <- best_hits_assemblyID_Annotated %>%\n",
    "\tcount(`Strain Category`, name = \"microcin\") %>%\n",
    "\tfull_join(strain_category_assemblies) %>%\n",
    "\tmutate(relative_microcin = microcin / total_assemblies) %>%\n",
    "\tmutate(odds = relative_microcin/(microcinAssemblyProportion)) %>%\n",
    "\tmutate(LOE = log10(odds)) %>%\n",
    "\tdrop_na() \n",
    "#microcin_strain_category <- microcin_strain_category %>%\n",
    "#\tmutate(assembly_freq = total_assemblies#/sum(total_assemblies)\n",
    "\n",
    "microcin_strain_category\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tChi-squared test for given probabilities\n",
      "\n",
      "data:  microcin_strain_category$total_assemblies\n",
      "X-squared = 241.17, df = 6, p-value < 2.2e-16\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "#Chi square present in paper before figure 5\n",
    "\n",
    "microcin_strain_category_chisq <- chisq.test(microcin_strain_category$total_assemblies)\n",
    "microcin_strain_category_chisq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tChi-squared test for given probabilities\n",
      "\n",
      "data:  microcin_strain_category$microcin\n",
      "X-squared = 56.158, df = 6, p-value = 2.705e-10\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "freqs = microcin_strain_category$total_assemblies/sum(microcin_strain_category$total_assemblies)\n",
    "\n",
    "microcin_strain_category_freq_chisq <- chisq.test(x = microcin_strain_category$microcin, p=freqs)\n",
    "\n",
    "microcin_strain_category_freq_chisq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 8 × 6\n",
      "  `Strain Category`    microcin total_assemblies relative_microc…  odds      LOE\n",
      "  <chr>                   <int>            <int>            <dbl> <dbl>    <dbl>\n",
      "1 Bird_Faecal                42              185           0.227  0.970 -0.0134 \n",
      "2 Human_Extra-intesti…       46              104           0.442  1.89   0.276  \n",
      "3 Human_Faecal               17               72           0.236  1.01   0.00366\n",
      "4 Human_Intestinal           33              172           0.192  0.819 -0.0865 \n",
      "5 Mammal_Faecal              26              117           0.222  0.949 -0.0227 \n",
      "6 Poultry_Meat               94              283           0.332  1.42   0.152  \n",
      "7 Water                      26              280           0.0929 0.397 -0.402  \n",
      "8 Total                     284             1213           1.74   7.45  -0.0924 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "microcin_strain_category #%>%\n",
    "#\twrite_csv(\"data/supplemental/Touchon_categories_microcins.csv\")\n",
    "\n",
    "microcin_strain_category %>%\n",
    "    bind_rows(summarise_all(., ~if(is.numeric(.)) sum(.) else \"Total\"))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 7 × 10\n",
      "  group   odds_ratio p.value conf.low conf.high method alternative   p.adj label\n",
      "  <fct>        <dbl>   <dbl>    <dbl>     <dbl> <chr>  <chr>         <dbl> <chr>\n",
      "1 Bird_F…      0.970 9.28e-1    0.660     1.40  Fishe… two.sided   1   e+0 \"\"   \n",
      "2 Human_…      1.89  1.22e-3    1.27      2.77  Fishe… two.sided   4.28e-3 \"**\" \n",
      "3 Human_…      1.01  1   e+0    0.548     1.76  Fishe… two.sided   1   e+0 \"\"   \n",
      "4 Human_…      0.820 3.41e-1    0.535     1.23  Fishe… two.sided   5.96e-1 \"\"   \n",
      "5 Mammal…      0.949 9.11e-1    0.584     1.49  Fishe… two.sided   1   e+0 \"\"   \n",
      "6 Poultr…      1.42  1.19e-2    1.07      1.86  Fishe… two.sided   2.77e-2 \"*\"  \n",
      "7 Water        0.397 2.83e-6    0.250     0.608 Fishe… two.sided   1.98e-5 \"***\"\n",
      "# … with 1 more variable: hjust <dbl>\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "touchon <- microcin_strain_category %>%\n",
    "    select(c(`Strain Category`,total_assemblies,microcin)) %>%\n",
    "\trename(\"group\"=\"Strain Category\") %>%\n",
    "\trename(\"samples\"=\"total_assemblies\") %>%\n",
    "\trename(\"hits\"=\"microcin\")\n",
    "\n",
    "mydata <- touchon\n",
    "\n",
    "samples_tot <- sum(mydata$samples)\n",
    "hits_tot <- sum(mydata$hits)\n",
    "\n",
    "fisher_results <- touchon %>%\n",
    "  mutate(\n",
    "    fit = map2(\n",
    "      samples, hits,\n",
    "      ~fisher.test(cbind(c(samples_tot, .x), c(hits_tot, .y)))\n",
    "    ),\n",
    "    glance_out = map(fit, glance)\n",
    "  ) %>%\n",
    "  select(group, glance_out) %>%\n",
    "  unnest(cols = glance_out) %>%\n",
    "  mutate(\n",
    "    # adjust for multiple testing\n",
    "    p.adj = p.adjust(p.value, method = \"fdr\"),\n",
    "    # sort\n",
    "    group = fct_reorder(group, estimate),\n",
    "    # significance label\n",
    "    label = case_when(\n",
    "      p.adj < 0.001 ~ \"***\",\n",
    "      p.adj < 0.01 ~ \"**\",\n",
    "      p.adj < 0.05 ~ \"*\",\n",
    "      TRUE ~ \"\"\n",
    "    ),\n",
    "    # for label justification\n",
    "    hjust = ifelse(estimate < 1, 1.1, -0.1)\n",
    "  ) %>%\n",
    "  rename(\n",
    "    odds_ratio = estimate\n",
    "  )\n",
    "\n",
    "fisher_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAyAAAAHgCAMAAABq2fnHAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqqu8qrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8bLYXQAAAgAElEQVR4nO3cfWBU9Z3v8UHR1VavdcVWuyy9t/e2u9ve3m3Zbe0+3F7v9m53t05MSYaJJEbSSCAhiICWQCMYgUWpkKA0PnazbNVCpaAgjdZkNpUEMU0RV0Ca6mLAQBIzCXmaDJOH+d3fmTnnOw+/mcM5w5yHIZ/3H5kz5/F3fpkXmQmKgyGEkuawegAI2TkAQUglAEFIJQBBSCVbAWl6TKWaarWtJlWz1eoR8LakZxBpms/NOvffqnf0m7UcoOlm9E3cm6HXpK2APKa20T9o1jBUOj9u9Qh458bScpqRtJxlwqvzgDG9o++b0LCTppvRNXETPw49AIiuACQ+ADExANEUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQpkO5MVLIwAxPgDJ4ADE+AAkgwMQ4wOQDA5AjA9AMjgAMT4AyeAAxPgAJIMDEOMDkAwOQIwPQDI4ADE+AMngAMT4ACSDAxDjA5AMDkCMD0AyOAAxPgDJ4ADE+AAkgwMQ4wOQDA5AjA9AMjgAMT4AyeAAxPgAJIMDEOMDkAwOQIwPQDI4ADE+AMngAMT4ACSDuxggeUelr673tB5d4pR6POn2zkKVg4WNXXnhRwDRFICklqlAPD5eQHm6sjduu+8gS7ZtZW/UxnAAoisASa3QxO1wjIZvihaSdJFAYl/j+V3SGScT7hrapvKcAYjOACS1+MS9H3xlWnCkkzFaSFpiIKeK+ENhp7focffdbU8UVLQz1lKes6CJeYt3FLi309EKkJaycVb3ozVZ85tLmgq84V2l+Lso+Qi+rb2jwl3JL/Bc4dx1Pv7cQxvlkwOIrgAktdi58zfOyp5eeM1dbFJZSF4ckNBnCicBcb4w/oyzPrBpAxvObvZ5XEFv1taxo1kfKUeHP4PsZ+yfXzg5f4j/VOhx1Y7Iu0rbJSDyEfld/sL60UMF3YcLTvet2MmfRzbKR4SANM6evbAX2TGrX9lpqrf3Y88PZzi+tuWdyELyumtigLRJnykiP0Fu97Hjd0ywA6vZRDcbb3OOeZ2nGSult2DhzyD8j9P+O8sOSm+bepxDyq4KEPmI/K7mUr5mw57WeYcnJiZkIOGN8hEhIGODg9VBlUYH1baalH/M6hHwzgXScpoRzXta/cpOU9LEeS6f/ble6Z5oIVnjyd9i3dnpzWGsvYSxg6tZcGfJkk0cyBy+oTwChD6D1M4PhoDwI+RdX8/Pf1IyIB+R37XbVczbFdy31LXuIxlIeKN8BN5i6QpvsVKLT5zv5vzBWbn8lmghaUk+gxQyNpAVDeRQcR/zcyB8TSIgp+9asTcExMWUXaXVkgH5iPwuTxVf6hjo7mJD26pkIOGN8hEAoisASS1p4jz9rPWUdE+0kKzEQPqcRybrnNFAGhYNB+qcI/FAmvy882xyRePpO3pY/ocSEHlXaXs0kA8H85v8rblnfrmkd7j2Af48slE+AkB0BSCplZa/B9mzcM5Di6OBBDa6yxqrKuKBhD6kl7LdlYy9sJbV5h3gQORdpe1RQGrzTpxYnlv2Fhurnjd3bRd/3kgb5SMARFcAklr4T02MD0AyOBOAnKwJd0zXUZoDEE0BSGrhJ4jxAUgGByDGByAZHIAYH4BkcABifACSwQGI8QFIBgcgxgcgGRyAGB+AZHAAYnwAksEBiPEBSAYHIMYHIBkcgBgfgGRwAGJ8AJLBAYjxAUgGByDGByAZHIAYH4BkcABifACSwQGI8QFIBgcgxgcgGRyAGB+AZHAAYnwAksEBiPFNTSBqTXh1HjCmd/R9Exp20nQzAGJ4ABIfgJgYgGgKQMQABEAoABEDEAChAEQMQACEAhAxAAEQ6tL+Na9wdQChAERTAJLizQCI4QEIgFgYgGgKQFK8GQAxPAABEAsDEE0BSIo3AyCGByAAYmEAoikASfFmAMTwAARALAxANAUgKd4MgBgegACIhQGIpgAkxZsBEMMDEACxMADRFICkeDMAYngAAiAWBiCaApAUbwZADA9AAMTCAERTAJLizQCI4QEIgFgYgGgKQFK8GQAxPAABEAsDEE0BSIo3Yz6QUqfTOa/aF7uyKy/xzn7nSulhIKsy2ekARFMAkuLNWACkwTfyu6JnY1dKQFb2ijv7nXPO8YfXswUgyt4AoqlLH8gOx2jk6hkN5AD/8tzS2JUSkPwu6RKTMev9zjWv8Yd1awUgob0ZgGjsUgfyfvCVacGRTuXqGQ/kp+Xs3WWu5cfZqSL+rLCTA1mTNb+5pKnAW1vHWHu+fIN+Z/1a/hLL28uBdFS4K48y1lKes6BJ2rs9tAeAaOoSBzJ546zs6YXX3KVcPaOBePz+94ue6J/7a1+DeygChP9M6HHVjrC3FzL2k1p5Z7+zx+1jh6per2T+wvrRQwXdw9nNPo8rGP4J0jh79sJeZMdMBdL7seeHMxxf2/KOxTfdXZMOIPxDeu6G4fo1fHmlJxaIc4grzDvNio8RkKF1b7DqX3EgzaX8+YY9E91svM05FgYyNjhYHVRpdFBtq0n5x6weAe9cIC2nGdG8p6lA+PU8l8/+XC9dvW88XTeja+LG0/YWi38KkX5IVP88BOROBUiOtGXzrvaiYARI48OTBQMcyG5XMW9XcGfJkk0KEIa3WBq7xN9iMd/N+YOzcunqGf0WKwxE+mzBVjWeKpR+iUtvsaQtb674yU+UnTmQoby3VzIOxFPFn3cMHCru42sBRGeXOhDm6Wetp+jqlwCQ/rktgSb3YJ/zyGSdMwzkwzCQ87kF7ys7cyBs9eKXJCCD+U3+1twzDYuGA3XOEb53eA8A0dQlDyS2SwAI+4+lrmXHGduzcM5Di0NAavMOhICw9QtpZwnIfudZCQg7sTy37C0W2Ogua6yq4HufCO0BIJoCkBRvxo7/qUnN8zp2BhBNAUiKN2M/IIHT7o/YyZpwxy64O4BoCkBSvBn7AXl73nN6dgcQTQFIijdjPyA6AxBNAUiKNwMghgcgAGJhAKIpAEnxZgDE8AAEQCwMQDQFICneDIAYHoAAiIUBiKYAJMWbARDDAxAAsTAA0RSApHgzAGJ4AAIgFgYgmgKQFG8GQAwPQADEwgBEUwCS4s0AiOEBCIBYGIBoCkBSvBkAMTwAARALAxBNAUiKNwMghgcgAGJhAKIpAEnxZgDE8KYmELUmvDoPGNM7egChAERTACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQlnwa97k2wDExABEUwAiBiAAQgGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACJmKyB5R6Wvrve07l/ilHo1+Q5deeFHANEUgIhlNhCPj6fy0gIQXQGIWGYDOXiBHQBEVwAiZksgp4r4Q2Gnt+hx991tTxRUtDPWUp6zoIl5i3cUuLfT/jKQ8DbWUeGu5Cd45x73A93yOgDRFYCI2QtI6DOFk4A4Xxh/xlkf2LSBDWc3+zyuoDdr69jRrI+U/UOfQZbJ2/yF9aOHCrp73b8ZqK2U14WAvHHrrYu9KvX2qm2dSqVpIrSf5sUX03KaFNN0gfSPoqcmdSBt0meKyE+Q233s+B0T7MBqNtHNxtucY17nacZK6S1Y6DOIX97WXMrXbNiz+xHGfB55XQjIaGfnoxMq+QbUtprUaMDqEfD6z6flNEOa93zxxeTbAl6dlz2vd/R9WuZc083omrhAmt5i3dnpzWGsvYSxg6tZcGfJkk0cyBy+oTwCJPQWS96221XM2/XU9qh1eIulK7zFErPXWywFSCFjA1nRQA4V9zE/B8LXCEDkbZ4qvtwxsGsz/wmyvSW8DkB0BSBitgTS5zwyWeeMBtKwaDhQ5xxJCETeNpjf5G/NPXPWfXj46eXyOgDRFYCI2RII27NwzkOLo4EENrrLGqsqEgKRt7ETy3PL3mKsrdy1qlNeByC6AhAxWwExLADRFICIZSKQkzXhjmk+AkA0BSBimQhEfwCiKQARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQCkDEAARAKAARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAo04GoBSAmBiCaAhAxAAEQCkDEAARAKAARAxAAoQBEDEAAhAIQMQABEMokIC9SansBiIkBiKYARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQCkDEAARAKAARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQCkDEAARAKAARAxAAoQBEzOZAnLy7nmesszCyrvKgslQibXa+mvzwrrzwI4BoCkDE7A7kuG/o6B3NzHcwsi4KiMfHU3lpAYiuAETM7kBO8i///DPl1JPS1yggBxMdExWA6ApAxOwPJHiy6PfSW6yekqYC74FF82or4oG0lOcsaGKso8JdeZSxd+5xP9AtrwMQXQGImN2BSG0LfQbpcdWOnMxtHdnujP0Msmw4u9nncQX9hfWjhwq6e92/GaitlNeFgBzMylrSr1KfV22rSfX1WT0Cnjc9g7jQWSJAVM/Sq/eyekev6fuu6aS6Jq53azqBHPf5OpdvDwFxDrE6/qNgoiD2M4h/opuNtznHmkv5mg17dj/CmM8jrwsBGXzvvUfGVfINqG01Kd95q0fA6/en5TRDF9geAaK2V8Cr87J+vaPvC2jY6UI3E0rXxJ3flk4g0meQhtIQkBzGNu/iT++Pe4sV3FmyZJNzbLermLfrqe1R6/AWS1d4iyVm97dYEhDP/PBbLMa2P85f+8VxQA4V9zG/c8xTxZc7BnZt5j9BtreE1wGIrgBEzO5ATvj93fdtUYB8mPtb/89vjwPSsGg4UOccGcxv8rfmnjnrPjz89HJ5HYDoCkDE7A6E51rXrwBhbyy6Y9ujcUACG91ljVUV7MTy3LK3GGsrd63qlNcBiK4ARMzmQNIUgGgKQMQyEcjJmnDHNB8BIJoCELFMBKI/ANEUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYgACIJRJQLQFICYGIJoCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQymggL8andhYAMTEA0RSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICI2QCI37lSehjIqtS0e1ceLZY4pV698L4AoikAEbMFkDnn+MPr2fqBeHw8lZcWgOgKQMRsAWTNa/xh3Vr9QA5q3BdANAUgYrYAUr+Wv0Ly9nIgLeU5C5qYt+hx991tTxRUtEcvyhtFIOH1rKPCXXmUsXfucT/QHbcvgGgKQMRsAaTH7WOHql6vZMPZzT6PK+h1vjD+jLM+sGkDi1qUN8Z/Blkmr/cX1o8eKujudf9moLYyet/WgoJ7z6nU51XbalL9/VaPgOdNzyD6km0QgKidpb9X52V1T6Gm2016M7rPRDtv1Q1kaN0brPpXHMhENxtvc455b/ex43dMsAOrWdSivDH+M4hfXt9cytds2LP7EcZ8nuh9+1pbN46pNHJObatJ+fxWj4DXP5qW0wwl2yAAUTvLea/Oy47qHX3feQ07Jb2Z6HRNnH+bfiCND08WDHAgwZ0lSzZxIDmMtZcwdpADiSzKG4W3WPL63a5i3q6ntketw1ssXeEtlpgt3mINDeW9vZJxIIeK+/jTJEDkjQIQeb2nii93DOzazH+CbG+J3RdANAUgYvYAwlYvfkkC0rBoOFDnHEkMRN4oAJHXD+Y3+Vtzz5x1Hx5+enncvgCiKQARswmQ/c6zEpDARndZY1VFYiDyRgGIvJ6dWJ5b9hZjbeWuVZ1x+wKIpgBEzAZATAhANAUgYpkI5GRNuGOajwAQTQGIWCYC0R+AaApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQCkDEAARAKAARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQymggugIQEwMQTQGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAilH4jwi1t9qZ0aQEwMQDQFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYgACIBSAiAEIgFAAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIABCAYgYgAAIBSBiAAIgFICIAQiAUAAiBiAAQgGIGIAACAUgYlYDyTsqfXW9p/Xslb9Q2dhZGLeiKy/8CCCaAhCxDAaysjd208pe38G4vQFEVwAilsFA8rtiN8U/ZwCiMwARswuQU0X8obDTW/S4++62Jwoq2hlrKc9Z0MS8xTsK3Ntpfw5EXrMma357R4W7kp/gucK563z8uaeQdg8fDCC6AhAxy4E4QxEQ5wvjzzjrA5s2sOHsZp/HFfRmbR07mvWRsr8ERF6T3+UvrB89VNB9uOB034qd/Dn/DCJvlA8OAWktKLj3nEp9XrWtJtXfb/UIeF7dg7hIIGqn7u/VORbdU6jpdvvSdibaeasuIG0+XuQnyO0+dvyOCXZgNZvoZuNtzjGv8zRjpfQWTAIir8nvai7lazbsaZ13eGJiQgYS3igfHALS09Cw7rxKw+fUtprUsM/qEfD6dQ/iIoGonXrUq3MsPr2j945q2GlQy5l0TZxvmy4g0W+x7uz05jDWXsLYwdUsuLNkySYOZA7fUB4DRF6T37XbVczbFdy31LXuIxlIeKN8MN5i6QpvscQsf4ulAClkbCArGsih4j7m50D4mjgg8pr8Lk8VX+oY6O5iQ9uqZCDhjfLBAKIrABGzC5A+55HJOmc0kIZFw4E654gKkA8H85v8rblnfrmkd7j2Af48AkQ+GEB0BSBidgHC9iyc89DiaCCBje6yxqqK5EBq806cWJ5b9hYbq543d20Xf95IQOSDAURXACJmNRBzAhBNAYhYhgA5WRPuWGqHA4imAEQsQ4BcZACiKQARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABEDEAChAEQMQACEAhAxAAEQCkDEAARAKAARAxAAoQBEDEAAhAIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAo/UASpuk1dcEAxMQARFMAIgYgAEIBiBiAAAgFIGIAAiAUgIgBCIBQACIGIMYBucjfgU6B1GYPQEwMQOyZ2uwBiIkBiD1Tmz0AMTEAsWdqswcgJgYg9kxt9gDExADEnqnNHoCYGIDYM7XZAxATAxB7pjZ7AGJiAGLP1GYPQEwMQOyZ2uwBiIkBiD1Tmz0AMTEAsWdqswcgJgYg9kxt9gDExADEnqnNHoCYGIDYM7XZAxATAxB7pjZ7AGJiAGLP1GYPQEwMQOyZ2uwBiIkBiD1Tmz0AMTEAsWdqswcgJgYg9kxt9gDExADEnqnNHoCkpW0P8i/rFvEvz/wg+V4AYs/UZg9A0lLTPMaCebf3MvaDf1XWreyN38tAIDsco8kWrH752T+1iQWQtNTjPMPeL17lYRM5bcq6/K6o0UxKXw0D8n7wlWnBkc5ECwBy4dSmFkDSU1ET212zo5qdzBphLeU5C5rYmqz57R0V7sqjrKekqSA0zUYBmbxxVvb0wmvuSrDAAOTCqc0tgKSnR59kD3pOzGevLWHD2c0+jyvIf4L4C+tHDxV097hq+f29XVq6fEClfq/aVtXOvfHAjY7ZW48nWOBZ/fKzf6pz26v3m3FO5wFeLQf0pe1MdMatpgJ5dfnE3N6JuZ3bnmQT3Wy8zTnGgTSX8i0b9vQ4h/hjT0PDuvMqDZ9T23qBfnX512adTbxw3uqXn/1Tm9lRr85vhc+n8wDvqIadBrWcqV/PpX3bTAVy+nvvljD2UP09B1hwZ8mSTSEgu13FvF09OfJOhn0G8d2cPzgrN+EC3mJdOLWpxVus9BSct46L3PtgtpcdKu5j/hAQTxXf0jHQ45J3Mu63WJ5+1noq8QKAXDC1mQWQNLU+6w2OIetuxhoWDQfqnCMs/8PB/CZ/a+4ZE4CoZfXLz/6pzR6ApKmXnP3858id1YwFNrrLGqsqWG3eiRPLc8veYgBi89RmD0BMDEDsmdrsAYiJAYg9U5s9ADExALFnarMHICYGIPZMbfYAxMQAxJ6pzR6AmBiA2DO12QMQEwMQe6Y2ewBiYgBiz9RmD0BMDEDsmdrsAYiJAYg9U5s9ADExALFnarMHICYGIPZMbfYAxMQAxJ6pzR6AmBiA2DO12QMQEwMQe6Y2ewBiYgBiz9RmD0BMDEDsmdrsAYiJAYg9U5s9ADExALFnarMHICYGIPZMbfYAxMQAxJ6pzR6AmBiA2DO12QMQE7MGiJ7Oj1s9Aqbz+5w8Ta+pCwYgJgYgmgIQMQABEApAxAAEQCgAEQMQAKEARAxAAIQCEDEAARAKQMQABEAoABGbGkCefkylH61X22pS1VutHgGvanNaTrMlLWepflDnATU1Og94sFrDTppu5iE9E7f1p6HXpK2AqPbS/VaPwC7Na7V6BFF9dKvRV/h2R7rO5D6s/xgAybwAJMUubSBHfmH1COzSsx9aPYKo+jcbfYXqvnSd6elT+o/JHCAIWRCAIKQSgKBUG/AmWjToCladKQOA9FS6V3eFFw+W5i5919rRWFlkJvY7pTZZO5xt+xItGnKFh6XbPZeOMx1dNnf1WR0HZgCQZc8P7bwntNQzt3VoZ17A4vFYV2QmRnt6erqXpPBLmfTVsjFrn7ho0BWWHuA3HEzDmXrdLaN7F+g4k/2BvH/HBAsWnJAWD29kLJDVZfWIrCpqJqT21Vo5GNa2f8k+cdGgK7gu7r1W5Ez1a/mXxe9oP9T+QA5U8C+rPeEnY+11U/evQ2JmgvV932flYHgb9iVaNOIK/VlrXIsPpuNML6/gX+6t136g/YG8UsW/rN8TfvLx6py9qf+kzfBiZoI9/pKVY5EyD0jH/Uf8B3P+Mw1nOpXd5n/NuUP7gfYH0hT6c7NBeTrsmrKf0mNmwuv2WzoYZiaQUOt1vKyTn+nAQvfmikvqJ0h7fpAFC49Jiy8/zb8sfdXiAVlW1Eww9rOt1g6GmQmkvY1/2XgxPzKVM40NMRb8/iX1GYQtfiX4y0VB1naWvZvbNtw4p9PqAVlWZCYYK3/T6tGEX3ShwRgKhF/h2Jy2QFvexfx2RjlTT06bf8eyS+q3WKxrVX4F/y4U8Vv0LHEtbbN6PNYVNRP9WQNWjyb8oivaR4sGXuHNe1wrLuqtNZ2pZcG8R4Z0HJgBQBCyLgBBSCUAQUglAEFIJQBBSCUAQUilqQDk23+jY+cnZ8QvJOxIG2PXPJts64fOa07ruKbcBw7lL7BmbFM7e1TSTtJQZtao7XHhc6SrhOeS7oZSnzj7BSBxaQRSkM1YXkOyrT+Y2ZLCv10VCyT52aOSdpKGkhzIhU+j6UIaS3iuGCDqE2e/ACQuNSCByKte+j4nb9G3dVyRigWiOXUgcQWSwtX2T+IlPz55ApDULp3StS+6KQVkcv2fXfuNV6SF1Z+btere7Li1IwtnziytmRG18B/f+dQN35PfLN20877rO4aW/o+rPr8uyG5xOBwDn3o26uDoM/0D3xr6t2qU3elEdMJ/nf2JL/0Lf5xZ/c1PfL727G2f+tzzHEjD9/7wC1WT4ZeUdHbaLWog3/weYwcc32dsL/fEdwoNZeaWys9flyv/8x/R5wydZmLNF6//+2PyLdCIlbXSHjN//G3Hda6+2Dum0Sc+XrgnFj6X8px2lO4m+cTRpRlLcOxVz/Ev82+Tr60MJNG8GNSUArL4qo37ixx7Gbvn+qd2ffUPsuPW3vpftuz61jUzIgujn/mrnz312f8X3u2mv7h9l3/O9Y/uW+p4gfXl/EN3UPo+08HRZ+ov/NvuSWmFsrtyIjrhY1esqb93Wi1/eVyxxnPbtJnV9V+/avgDxw0LX14xbUUEiLJb9EDW3BBkDzu+wNj9nw5KO4WGMvMzBW/UXBX+vw1jzhk6TfF1P37+b/inotAt0IiVtSEg1+f/etP0xbF3rIw+8fHCPUnxc9FzupB0N8knTrm0lHhsBIh0bWUgiebFoKYSkI+mS//473e/ys5eyb/rZ6Znx679d8c+xs7/0YzIwm8djYy9fF/46Ju+wv8szXmSL33xB+F3Cvz7TAfHnIneYim7KydSHkduWMc3LPg0f3m4GPudYylj+x1HP3C4+dr7r+pXgNBu0QNpdhxjWX/n6GHfLAgrCr3F+nO+pfBb4atGn1Pa4/eX/Zzf7pVbQ7dAg6S1ISB/zm/O+dexd6yMPvHx8fcUip9LeR6ZDelukk+ccmkp8dgIEH5tZSAJ58WgphKQfY5u/nX7ZYG9jmG+8PXs2LUPXy89vXcGowXvtf+zjv4H/5tWhlqaGPQAAARDSURBVB6Gjzx7xX30faaDY84U9RkkvLtyIuWx1fGm1+v9qeM0m/kIfwcj/Sn9O8c7Hzhe5tuOOloUILRb9EDGr/0xu3Hv1btHr/y3KCCrpRHL9xl9TmmPf5suvckf8odugQZJa0NAVvHl+26JvWNl9ImPj7+nUPxcyvPIbIQ/gySbOOXSLOpcka0RIPzaykASzotBTSUgT14m/WfODY5T266Wnt6WHbv23i9LT7fMYLTADjuvcnxV/l/4btrCv7R8Zdpn//Gzke8zHRxzJgJCuysnkh93OsK9y2Y+Kr2YfyEDeYsfc86xWwES2S16IE7376ed+7/Lfu3oigJSzaKARJ1T2mP9TSxyCzRIWhsC8iMWfpVGX0gZfeLj4++JziU/j+wo3U3yiaNLSwnHRoDwaysDSTwvxjSVgOzj70sYe26a/xcO6V/LvyU7du2P/lB6+sMZjBZ4o69/5/Lfhfa7qYax/ivK+M63RP9BKB8ccyYFSGT3yIlCj//u+FgeWywQ6SfIcUebAiSyW/RAHrup7sts7V+u/18sCoj0W6xkQJ69UvpAdKQ9dAs0SFobAsIPkV+ldCEafeLj4+8pdOnQbxbCzyM78rtRmbioSyc4NgQk+7bw9CsDSTwvxjSVgJyeLv2sz/oKO3n5Dsa6rsyOXfuGYz9/Yf3JjMjCz7/AIZ10hP8XRuk79LrjA8b8N0e+z3RwzJkUILS7ciLl8eM/+Anf/MDfxQOZx9eu+tSIAoR2ixnICce3Sphn+l9L/3qFJiDvSJ+p/DdsDt0CDZLWRgOJvhCNPvHx8fcUujQ/l/I8Mhv8blQmLhqIeOwnHmTM92kZiDKQxPNiTFMCyJ++LHWclV396GsLpD+mF97wL3u/8Udzwptp7a3XPb73H/94RmTh91d+d//P/n5G+FeQ0neo4wpX895vfPKfvKzoS7+dkF5ZdHDMmWQgtLtyIjrhyivX198/bWs8kGnlv1x52cOR32Ipu8UMhM10bGe+KxzS37ZJO0lDUQXCXDc89XrWdZ2hW4gMUlkbDST6QpGbTXi8cE8sPB56Thfid6MycdFAxGNvuXn7r79zmQyEBpJ4XgxpSgAJv2OtYBMP/ek1X5f+17KxZZ/5k+e/UxjeTGuHS/745pL9M6IWXpn9yRn/JP/tXeg7tOOLn/z6K9uvX8Oa/tu1g6G/YlAOjjmT8hmEdldOpDwGH/3yJ/5M+sVOLJD6/3Pdl6TLKEBot+iBsCLH+4x982rpTY60kzQUdSCB+/77td/6jXwLNEhlbcxbrOgL0egTHx9/T0wej/KcdpTuJvnExbzFEo59739f7fiLcgWIMpDE82JIUwGI0PCT/EN18L9usHocSEPBHksvPyWBBGd994P+h640+BeE6FJoSgJhb3/NMe0Lr1k9CpQBTU0gjA0OWz0ClBFNVSAIaQpAEFIJQBBSCUAQUglAEFLp/wMuamGWHd/eAAAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -w 800\n",
    "\n",
    "ggplot(fisher_results, aes(odds_ratio, group)) +\n",
    "    geom_col() +\n",
    "    geom_text(aes(label = label, hjust = hjust), size = 8, vjust = 0.8) +\n",
    "    scale_x_log10(breaks=seq(0.3,2,0.4), limits = c(0.3, 2)) +\n",
    "    geom_bar(stat = \"identity\", position = \"identity\", fill = \"#aabbcaff\") +\n",
    "        ylab(NULL) + \n",
    "        xlab(\"Log odds ratio of assemblies with microcins across isolation sources\") +\n",
    "        theme_bw() +\n",
    "        theme(text = element_text(size = 16))\n",
    "\n",
    "#ggsave(\"plots/05_ecoli_microcin_environment_updated_hmm_final.svg\")"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "184da163bdcbc68ba00985bd6e7412525ff7507797be2878afd169be169c48f1"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
